openedx/features/portfolio_project/views.py

```diff
@@ -30,16 +33,31 @@ def get(self, request, course_id, **kwargs):
         """
         Displays a generic tab for the specified course.
         """
+        self.course_id = course_id
+
         return super(GenericTabView, self).get(request, course_id, 'courseware', **kwargs)
 
     def render_to_fragment(self, request, course=None, tab=None, **kwargs):
         """
         Render out the bootstrap page.
         """
+        course_key = CourseKey.from_string(self.course_id)
+        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=True)
+        enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
+
+        upgrade_price = None
+        upgrade_url = None
+
+        if enrollment and enrollment.upgrade_deadline:
+            upgrade_url = EcommerceService().upgrade_url(request.user, course_key)
+            upgrade_price = get_cosmetic_verified_display_price(course)
+
         context = {
             'course': course,
             'user': request.user,
             'tab_name': tab,
+            'upgrade_url': upgrade_url,
+            'upgrade_price': upgrade_price,
```

Comment:
@jaebradley @MatthewPiatetsky: Do we know why the cosmetic price is not appearing in the `user-metadata`?  Does it make more sense to enhance that with `'upgrade_price_ui` (or something like that) if we need to support both?
Also, do we know why all the other metadata works, but upgrade_url does not work?  Does that need to be fixed in the original code?

Once this all gets worked out, we should update the following docs with any caveats if there need to be any:
https://openedx.atlassian.net/wiki/spaces/EXP/pages/199885286/Optimizely+Tips+and+Tricks#OptimizelyTipsandTricks-UpsellExperiments